resources/gui/js/api.js

```diff
@@ -0,0 +1,543 @@
+   */
+  static loadTemplateContent(id) {
+    id = `${id}-template`;
+
+    if (this._templates.has(id)) {
+      return this._templates.get(id).cloneNode(true).firstElementChild;
+    }
+
+    const template = document.getElementById(id);
+
+    if (template === null) {
+      console.error(`Template '#${id}' not found.`);
+      return false;
+    }
+
+    const { content } = template;
+    this._templates.set(id, content);
+
+    return content.cloneNode(true).firstElementChild;
+  }
+
+  /**
+   * Clear the content of an element if it exists
+   *
+   * @param element {string|HTMLElement} Element or ID
+   * @return {void}
+   */
+  static clearHtml(element) {
+    if (typeof element === 'string') {
+      element = document.getElementById(element);
+    }
+
+    if (element !== null && element instanceof HTMLElement) {
+      while (element.firstChild !== null) {
+        element.removeChild(element.firstChild);
+      }
+    } else {
+      console.warn('Element could not be cleared.');
+    }
+  }
+
+  /**
+   * Initialize a noUiSlider
+   *
+   * @param id {string}
+   * @param min {number}
+   * @param max {number}
+   * @param step {number}
+   * @param start {number[]}
+   */
+  static initSlider(id, min, max, step, start) {
+    const slider = document.getElementById(id);
+
+    if (typeof noUiSlider === 'undefined') {
+      console.error('\'noUiSlider\' is not defined.');
+      return;
+    }
+
+    noUiSlider.create(slider, {
+      start,
+      connect: (start.length === 1 ? 'lower' : true),
+      step,
+      range: { min, max },
+      format: {
+        to(value) {
+          return Format.beautifyNumber(value);
+        },
+        from(value) {
+          return Number(parseFloat(value));
+        },
+      },
+    });
+
+    slider.noUiSlider.on('slide', (values, handle, unencoded) => {
+      if (Array.isArray(unencoded)) {
+        CosmoScout.callNative(id, unencoded[handle], handle);
+      } else {
+        CosmoScout.callNative(id, unencoded, 0);
+      }
+    });
+  }
+
+  /**
+   * Set a noUiSlider value
+   *
+   * @param id {string} Slider ID
+   * @param value {number} Value
+   */
+  static setSliderValue(id, ...value) {
+    const slider = document.getElementById(id);
+
+    if (slider !== null && typeof slider.noUiSlider !== 'undefined') {
+      if (value.length === 1) {
+        slider.noUiSlider.set(value[0]);
+      } else {
+        slider.noUiSlider.set(value);
+      }
+    } else {
+      console.warn(`Slider '${id} 'not found or 'noUiSlider' not active.`);
+    }
+  }
+
+  /**
+   * Global entry point to call any method on any registered api
+   *
+   * @param api {string} Name of api
+   * @param method {string} Method name
+   * @param args {string|number|boolean|Function|Object} Arguments to pass through
+   * @return {*}
+   */
+  static call(api, method, ...args) {
+    if (method !== 'setUserPosition' && method !== 'setNorthDirection' && method !== 'setSpeed' && method !== 'setDate') {
```

Comment:
This can be removed, I guess :smiley:.

Reply:
Fully removed `call` and replaced it with direct object accesses.